### PR TITLE
fix: call presentPaywallIfNeeded when needed

### PR DESCRIPTION
**Pull request by danjocha**
Thank you for contributing to Purchases. Before pressing the "Create Pull Request" button, please provide the following:

  - [x] A description about what and why you are contributing, even if it's trivial.
  - Call the `presentPaywallIfNeeded` method when the `requiredEntitlementIdentifier` is not empty

  - [x] The issue number(s) or PR number(s) in the description if you are contributing in response to those.
[https://github.com/RevenueCat/purchases-flutter/issues/1065](url)
  - [ ] If applicable, unit tests.

### CODE REVIEW

File: purchases_ui_flutter/ios/Classes/PurchasesUiFlutterPlugin.swift [hunk 1]
```diff
@@ -106,13 +106,19 @@ public class PurchasesUiFlutterPlugin: NSObject, FlutterPlugin {
             if let requiredEntitlementIdentifier {
```
Comment (vegaro) [inline purchases_ui_flutter/ios/Classes/PurchasesUiFlutterPlugin.swift:106]: I think we can remove this `if let`

File: purchases_ui_flutter/ios/Classes/PurchasesUiFlutterPlugin.swift [hunk 2]
```diff
@@ -106,13 +106,19 @@ public class PurchasesUiFlutterPlugin: NSObject, FlutterPlugin {
...
                 options[PaywallProxy.PaywallOptionsKeys.requiredEntitlementIdentifier] = requiredEntitlementIdentifier
             }
-            if let offeringIdentifier {
```
Comment (vegaro) [inline purchases_ui_flutter/ios/Classes/PurchasesUiFlutterPlugin.swift:109]: we still need this

**Review by vegaro — CHANGES_REQUESTED** [review]
Thanks for opening, that makes sense, good catch

Let us know if you prefer us taking care of the changes and we can get the bugfix released soon

**Comment by danjocha** [comment]
@vegaro I just pushed another commit. I removed the incorrect `if let`, let me know if this code works or if you want to take it from here. 

Happy to contribute wherever I can

**Review by vegaro — APPROVED** [review]

**Comment by vegaro** [comment]
Thanks @danjocha I will take care of merging this

**Pull request merged**